tests/cases/compiler/-da/divergentAccessorsTypes-56894-a.ts

```diff
@@ -0,0 +1,27 @@
+// @strict: true
+// @declaration: true
+// @target: esnext
+
+declare const id:<T>()=>T
+
+function f2(x: {f:((a:1|2)=>1|2)} | {f:((a:2|3)=>2|3)} ){
+    const r = x.f(id())
+//                ^ const id: <2>() => 2,
+//        ^ const r: 1 | 2 | 3
+}
+
+function f3(f: ((a:1|2)=>1|2) | ((a:2|3)=>2|3) ){
+    const r = f(id())
+//              ^ const id: <2>() => 2,
+//        ^ const r: 1 | 2 | 3
+}
+
+
+function f4(x: {set d(a:1|2); get r():1|2} | {set d(a:2|3); get r():2|3} ){
+//           ^ (parameter) x: { d: 1 | 2; readonly r: 1 | 2; } | { d: 2 | 3; readonly r: 2 | 3; }
+    x.d = id();
+//        ^ const id: <1 | 2 | 3>() => 1 | 2 | 3      (5.3.2)
+//        ^ const id: <2>() => 2      (expecting, if a setter was treated as a function)
+    x.r;

```

Comment:
Was expecting pull #56895 to change this behavior, because therein it say it fixes #5689.  But no change occurred as far as I can see.

Reply:
Getters are treated like properties so a union of getters returns the union of their return types. I think this just works as expected. Even if we treat them as functions - it still all checks out, return types are unionified:
```ts
declare const fn: (() => 1 | 2) | (() => 2 | 3)
const result = fn()
//    ^? const result: 1 | 2 | 3
```

Reply:
My comment and the subject of #56894 is setters.  The point of the result diff is that valid domain has not narrowed by your fix. For more clarity see #56922.
```
function f4(x: {set d(a:1|2); get r():1|2} | {set d(a:2|3); get r():2|3} ){
//           ^ (parameter) x: { d: 1 | 2; readonly r: 1 | 2; } | { d: 2 | 3; readonly r: 2 | 3; }
    x.d = id();
//        ^ const id: <1 | 2 | 3>() => 1 | 2 | 3      (5.3.2)
//        ^ const id: <2>() => 2      (expecting, if a setter was treated as a function)
```